src/Compilers/CSharp/Portable/Symbols/Source/SourceLocalSymbol.cs

```diff
@@ -580,5 +598,58 @@ protected override TypeSymbol InferTypeOfVarVariable(DiagnosticBag diagnostics)
                 }
             }
         }
+
+        /// <summary>
+        /// Symbol for a deconstruction local that might require type inference.
+        /// For instance, local `x` in `var(x, y) = ...` or `(var x, int y) = ...`.
+        /// </summary>
+        private class PossiblyImplicitlyTypedDeconstructionLocalSymbol : SourceLocalSymbol
+        {
+            public PossiblyImplicitlyTypedDeconstructionLocalSymbol(
+                Symbol containingSymbol,
+                Binder binder,
+                TypeSyntax typeSyntax,
+                SyntaxToken identifierToken,
+                LocalDeclarationKind declarationKind)
+            : base(containingSymbol, binder, RefKind.None, typeSyntax, identifierToken, declarationKind)
+            {
+            }
+
+            protected override TypeSymbol InferTypeOfVarVariable(DiagnosticBag diagnostics)
+            {
+                // Try binding immediately enclosing deconstruction-declaration (the top-level VariableDeclaration), this should force the inference.
+
+                var topLevelVariableDeclaration = (VariableDeclarationSyntax)IdentifierToken.
+                                                                    Parent. // VariableDeclaratorSyntax
+                                                                    Parent; // VariableDeclarationSyntax
+                while (true)
+                {
+                    var parent = topLevelVariableDeclaration.Parent;
+                    if (parent.Kind() != SyntaxKind.VariableDeconstructionDeclarator) { break; }
```

Comment:
Fixed the formatting for braces.

Good question on null parent. I only create such local symbols when I am in a valid context. Do you think I should have additional guards?
If so, does `PossiblyImplicitlyTypedOutVarLocalSymbol` have the same problem?